lib/ecmascript.ts

```diff
@@ -3850,397 +4063,366 @@ export const ES = ObjectAssign({}, ES2020, {
       0,
       0,
       0,
-      0
-    );
-    const dayLengthNs = dayEnd.subtract(dayStart);
-
-    if (timeRemainderNs.subtract(dayLengthNs).multiply(direction).geq(0)) {
-      ({ years, months, weeks, days } = ES.AddDuration(
-        years,
-        months,
-        weeks,
-        days,
-        0,
-        0,
-        0,
-        0,
-        0,
-        0,
-        0,
-        0,
-        0,
-        direction,
-        0,
-        0,
-        0,
-        0,
-        0,
-        0,
-        relativeTo
-      ));
-      timeRemainderNs = ES.RoundInstant(timeRemainderNs.subtract(dayLengthNs), increment, unit, roundingMode);
-      ({ hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = ES.BalanceDuration(
-        0,
-        0,
-        0,
-        0,
-        0,
-        0,
-        timeRemainderNs.toJSNumber(),
-        'hour'
-      ));
+      0,
+      relativeTo
+    ));
+    timeRemainderNs = RoundInstant(timeRemainderNs.subtract(dayLengthNs), increment, unit, roundingMode);
+    ({ hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = BalanceDuration(
+      0,
+      0,
+      0,
+      0,
+      0,
+      0,
+      timeRemainderNs.toJSNumber(),
+      'hour'
+    ));
+  }
+  return { years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds };
+}
+
+export function RoundDuration(
+  years,
+  months,
+  weeks,
+  days,
+  hours,
+  minutes,
+  seconds,
+  milliseconds,
+  microseconds,
+  nanoseconds,
+  increment,
+  unit,
+  roundingMode,
+  relativeTo = undefined
```

Comment:
Should this be `relativeTo?`? Is there any difference between marking as optional in TS and giving it a default value of `undefined`?

Reply:
OK sounds good. I know that in the public API, relativeTo can be a PlainDateTime or a ZonedDateTime (or undefined) but not sure how that shakes out down here in the implementation.